worker/undertaker/undertaker.go

```diff
@@ -4,28 +4,32 @@
 package undertaker
 
 import (
-	stdcontext "context"
+	"context"
 	"fmt"
 	"time"
 
 	"github.com/juju/clock"
 	"github.com/juju/errors"
 	"github.com/juju/worker/v3/catacomb"
+	"gopkg.in/retry.v1"
```

Comment:
FWIW we're trying to move away from this and `juju/retry` is the preferred lib.

Reply:
Happy to loop back and use juju/retry. I feel like "gopkg.in/retry.v1" works better here, but will propose a seperate PR to drop it.